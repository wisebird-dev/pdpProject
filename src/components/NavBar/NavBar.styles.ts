import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles(({ palette, spacing, breakpoints, zIndex }) => ({
    navBar: {
        backgroundColor: palette.background.default,
        position: 'fixed',
        top: 0,
        left: 0,
        width: '100%',
        display: 'grid',
        gridColumnGap: spacing(1),
        gridTemplateColumns: 'auto auto auto auto', // the last column will be the cart component that will be created in the future
        justifyContent: 'flex-start',
        alignItems: 'center',
        zIndex: zIndex.appBar,
    },
    siteName: {
        [breakpoints.down('xs')]: {
            display: 'none',
        },
    },
}));

export default useStyles;
