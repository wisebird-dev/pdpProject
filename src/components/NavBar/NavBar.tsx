import React from 'react';

import { Typography } from '@components';

import useStyles from './NavBar.styles';

const NavBar: React.FC = () => {
    const classes = useStyles();

    return (
        <div className={classes.navBar}>
            <Typography component="h1" variant="h4" color="textSecondary">
                nav bar element
            </Typography>
        </div>
    );
};

export default NavBar;
