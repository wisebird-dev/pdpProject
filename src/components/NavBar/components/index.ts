export { Search } from './Search';
export { Sidemenu } from './Sidemenu';
export { SiteName } from './SiteName';
