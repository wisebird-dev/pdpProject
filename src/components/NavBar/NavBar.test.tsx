import React from 'react';

import { render } from '@testUtils';

import NavBar from './NavBar';

describe('NavBar suite', () => {
    it('renders correctly', () => {
        const {
            container: { firstChild },
        } = render(<NavBar />);

        expect(firstChild).toMatchSnapshot();
    });
});
