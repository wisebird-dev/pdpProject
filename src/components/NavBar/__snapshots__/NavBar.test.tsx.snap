// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar suite renders correctly 1`] = `
<div
  class="makeStyles-navBar-1"
>
  <div>
    <button
      aria-label="toggle menu"
      class="MuiButtonBase-root MuiIconButton-root"
      data-testid="toggle-sidemenu-button"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiIconButton-label"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-colorSecondary"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
          />
        </svg>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <div
      class="PrivateSwipeArea-root-14 PrivateSwipeArea-anchorLeft-15"
      style="width: 20px;"
    />
  </div>
  <h1
    class="MuiTypography-root makeStyles-siteNameMobile-7 MuiTypography-h5 MuiTypography-colorTextSecondary"
  >
    P
  </h1>
  <h1
    class="MuiTypography-root makeStyles-siteName-6 MuiTypography-h5 MuiTypography-colorTextSecondary"
  >
    PDP project
  </h1>
  <div
    aria-expanded="false"
    class="makeStyles-search-8"
    data-testid="autocomplete"
    role="combobox"
  >
    <div
      class="makeStyles-searchIcon-9"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
        />
      </svg>
    </div>
    <div
      aria-autocomplete="list"
      aria-expanded="true"
      autocapitalize="none"
      class="MuiInputBase-root makeStyles-inputRoot-10 Mui-focused"
      role="combobox"
      spellcheck="false"
    >
      <input
        aria-label="search"
        autocomplete="off"
        class="MuiInputBase-input makeStyles-inputInput-11"
        data-testid="search-input"
        id="search-input"
        placeholder="Search..."
        type="text"
        value=""
      />
    </div>
  </div>
</div>
`;
